### ISSUE TITLE
Add Catalyst components, support building an separate npm package for catalyst

### ISSUE THREAD

**Issue by mikemurray** [reactioncommerce/reaction-component-library#418]
## Issue

tl;dr Add Catalyst UI components to this repo as a separate package, but sharing the docs site.

The Catalyst components currently live in Reaction here: https://github.com/reactioncommerce/reaction/tree/master/imports/client/ui. These components are only for the administration app, not for use on storefronts, so they need to be separate packages.

## Possible Solution

- Create a `packages` directory
- Move `package` into `packages`, rename to `components`
- Add a `catalyst-components` directory
- Add a single catalyst component, probably the `Button` component
- Fix build to support publishing two separate npm packages `@reactioncommerce/components` & `@reactioncommerce/catalyst-components`
- Add a section to the docs for catalyst components

**This shouldn't be a breaking change for the original `@reactioncommerce/components`, nor should a new package need to be released for it.**
